Controller and Services Solicitude

smartinezvila29 opened the pull request:
TODO: Conectar con aula.

smartinezvila29 committed:
avance

smartinezvila29 committed:
Merge branch 'develop' into entity-controller-repository-Order

# Conflicts:
#	pom.xml
Status: resolved

smartinezvila29 committed:
avance controllers y services

Delete Order
Create Solicitude
Avance controllers y services

smartinezvila29 committed:
Finalizacion CRUD Pedido

OlivetoIgn commented on src/main/java/com/unla/aulas/controller/SolicitudeController.java:
```diff
@@ -0,0 +1,48 @@
+    @Autowired
+    SolicitudeService solicitudeService;
+
+    @GetMapping("query")
```
Por q query? Aca tendria q ser el path base de solicitude

OlivetoIgn commented on src/main/java/com/unla/aulas/controller/SubjectController.java:
```diff
@@ -0,0 +1,48 @@
+    @Autowired
+    SubjectService subjectService;
+
+    @PostMapping("save")
```
Aca tampoco el save .. con el path solo va

OlivetoIgn commented on src/main/java/com/unla/aulas/dto/DepartmentDto.java:
```diff
@@ -0,0 +1,31 @@
+
+    public DepartmentDto(){}
+
+    public String getDepartment() {
+        return department;
+    }
+
+    public void setDepartment(String department) {
+        this.department = department;
+    }
+
+    public DepartmentDto(int id, String department) {
+        this.department = department;
+        this.id = id;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public void setId(int id) {
+        this.id = id;
```
LA idea de usar lombok es no hacer setters ni getters

OlivetoIgn commented on src/main/java/com/unla/aulas/dto/SubjectDto.java:
```diff
@@ -0,0 +1,54 @@
+    private String subject;
+    private CarrerDto carrer;
+
+    public int getId() {
+        return id;
+    }
+
+    public void setId(int id) {
+        this.id = id;
+    }
+
+    public int getSubjectCode() {
+        return subjectCode;
+    }
+
+    public void setSubjectCode(int subjectCode) {
+        this.subjectCode = subjectCode;
+    }
+
+    public String getSubject() {
+        return subject;
+    }
+
+    public void setSubject(String subject) {
+        this.subject = subject;
+    }
+
+    public CarrerDto getCarrer() {
+        return carrer;
+    }
+
+    public void setCarrer(CarrerDto carrer) {
+        this.carrer = carrer;
+    }
```
Si usamos lombok no hay necesidad de setters ni getters

smartinezvila29 replied:
Buenisimo, desconocia. Fixed

OlivetoIgn commented on src/main/java/com/unla/aulas/entity/SolicitudeEntity.java:
```diff
@@ -0,0 +1,61 @@
+    private LocalDate endDate;
+
+    @Override
+    public boolean equals(Object o) {
```
No hay necesidad de este equals con lombok

OlivetoIgn replied:
https://projectlombok.org/features/EqualsAndHashCode

OlivetoIgn commented on src/main/java/com/unla/aulas/entity/SubjectEntity.java:
```diff
@@ -0,0 +1,34 @@
+    private CarrerEntity carrer;
+
+    @Override
+    public boolean equals(Object o) {
```
Lo mismo aca

smartinezvila29 replied:
Fix

OlivetoIgn commented on src/main/java/com/unla/aulas/security/SecurityConfig.java:
```diff
@@ -31,7 +31,7 @@ public SecurityConfig(UserEntityAuthService userEntityAuthService, BCryptPasswor
         http.cors().and().csrf().disable().authorizeRequests()
                 .antMatchers(HttpMethod.POST, SIGN_UP_URL).permitAll()
-                .anyRequest().authenticated()
+                .anyRequest().permitAll() //authenticated
```
Esto es para testear local .. al repo tiene que ir con los endpoints protegidos

smartinezvila29 replied:
Fixed

OlivetoIgn commented on src/main/java/com/unla/aulas/controller/CarrerController.java:
```diff
@@ -0,0 +1,48 @@
+    @PostMapping("save")
+    public String saveCarrer(@RequestBody CarrerDto json){
+        boolean resp = carrerService.saveCarrer(json);
+        if(!resp){
```
Si hay algun problema en el servicio va a romper ahi adentro y esto nunca se va a considerar ... hay q validar en el servicio las excepciones

smartinezvila29 replied:
Esta considerado en el service, si rompe devuelve false, y al postman no se pudo crear. Al conectarlo con el front podemos implementar codigos https junto con la respuesta (200, 404, 500)

smartinezvila29 committed:
fix comments

Fix comments
Sube solicitude_code como codigo pedido. En caso de usarlo, se implementa logica

smartinezvila29 committed:
Se saca turno de pedido

turno en pedido no tiene sentido ya que esta en espacio
